libsweep/src/sweep.cc

```diff
@@ -49,54 +51,49 @@ void sweep_error_destruct(sweep_error_s error) {
   delete error;
 }
 
-sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) {
+sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) try {
   SWEEP_ASSERT(error);
-
   return sweep_device_construct(port, 115200, error);
+} catch (const std::exception& e) {
+  *error = sweep_error_construct(e.what());
```

Comment:
I'm pretty sure that isn't legal: `e.what()` returns a pointer to a string that doesn't exist after the destruction of the exception (at the end of the catch clause). `sweep_error_construct()` only copies that pointer, but doesn't copy the string, so when you try to access it later you read from freed memory.